### PR TITLE
Fixes devices components

### PR DESCRIPTION
**Pull request by LKuemmel**

### CODE REVIEW

File: modules/speicher_fronius/main.sh [hunk 1]
```diff
@@ -18,11 +17,11 @@ fi
...
 openwbDebugLog ${DMOD} 2 "Speicher IP: ${wrfroniusip}"
 
-bash "$OPENWBBASEDIR/packages/legacy_run.sh" "modules.fronius.device" "bat" "${wrfroniusip}" "${froniuserzeugung}" "0" "none" "" >>$MYLOGFILE 2>&1
+bash "$OPENWBBASEDIR/packages/legacy_run.sh" "modules.fronius.device" "bat" "${wrfroniusip}" "${froniuserzeugung}" "0">>"$MYLOGFILE" 2>&1
 ret=$?
 
 openwbDebugLog ${DMOD} 2 "RET: ${ret}"
 
-speicherleistung=$(<${RAMDISKDIR}/speicherleistung)
+speicherleistung=$(<"${RAMDISKDIR}"/speicherleistung)
```
Comment (benderl) [inline modules/speicher_fronius/main.sh:25]: Auch wenn die Syntax korrekt ist, würde ich für eine bessere Lesbarkeit das schließende " ans Ende verschieben.
```suggestion
speicherleistung=$(<"${RAMDISKDIR}/speicherleistung")
```

File: packages/modules/json/device.py [hunk 2]
```diff
@@ -17,9 +18,9 @@ def get_default_config() -> dict:
         "type": "json",
         "id": 0,
         "configuration": {
-            "ip_address": None
-            # ToDo: add protocol
-            # ToDo: add port
+            "protocol": "http",
+            "domain": None,
+            "port": ""
```
Comment (benderl) [inline packages/modules/json/device.py:23]: Das ist eine numerische Angabe.
```suggestion
            "port": 80
```

**Review by benderl — CHANGES_REQUESTED** [review]
Siehe inline Kommentare.

**Comment by ToboliGermany** [comment]
Da scheint etwas noch nicht zu passen.... Ich bekomme keine Daten mehr nach dem Update:

2022-05-17 07:10:22: PID: 2381: legacy run server: Received command ["modules.fronius.device","inverter","192.168.200.252","0","0","none","1"]
2022-05-17 07:10:24: PID: 2381: root: Fronius IP-Adresse: 192.168.200.252
2022-05-17 07:10:24: PID: 2381: root: Komponente Fronius Wechselrichter auslesen.
2022-05-17 07:10:24: PID: 2381: root: Fronius Wechselrichter: FaultState FaultStateLevel.ERROR, FaultStr <class 'KeyError'> 'ip_address', Traceback:
Traceback (most recent call last):
File "/var/www/html/openWB/packages/modules/fronius/device.py", line 111, in read_legacy
total_power = inverter1.read_power()
File "/var/www/html/openWB/packages/modules/fronius/inverter.py", line 40, in read_power
'http://' + self.device_config["ip_address"] + '/solar_api/v1/GetPowerFlowRealtimeData.fcgi',
KeyError: 'ip_address'

2022-05-17 07:10:24: PID: 2381: legacy run server: Completed running command in 2.08s: ["modules.fronius.device","inverter","192.168.200.252","0","0","none","1"]
2022-05-17 07:10:28: PID: 2381: legacy run server: Received command ["modules.fronius.device","counter_s0","192.168.200.252","1","1","none"]
2022-05-17 07:10:28: PID: 2381: root: Fronius IP-Adresse: 192.168.200.252
2022-05-17 07:10:28: PID: 2381: legacy run server: Unhandled exception
Traceback (most recent call last):
File "/var/www/html/openWB/packages/legacy_run_server.py", line 48, in redirect_stdout_stderr_exceptions_to_log
yield
File "/var/www/html/openWB/packages/legacy_run_server.py", line 87, in handle_connection
self.__callback(read_all_bytes(connection))
File "/var/www/html/openWB/packages/legacy_run_server.py", line 109, in handle_message
importlib.import_module(parsed[0]).main(parsed[1:])
File "/var/www/html/openWB/packages/modules/fronius/device.py", line 120, in main
run_using_positional_cli_args(read_legacy, argv)
File "/var/www/html/openWB/packages/helpermodules/cli/_run_using_positional_cli_args.py", line 34, in run_using_positional_cli_args
args.RUN(args)
File "/var/www/html/openWB/packages/helpermodules/cli/_run_using_positional_cli_args.py", line 19, in
parser.set_defaults(RUN=lambda args: function(*[getattr(args, argument_name) for argument_name in arg_spec.args]))
File "/var/www/html/openWB/packages/modules/fronius/device.py", line 116, in read_legacy
raise Exception("illegal component num " + str(num) + ". Should be an int if it is an inverter.")
Exception: illegal component num None. Should be an int if it is an inverter.
2022-05-17 07:10:31: PID: 2381: legacy run server: Received command ["modules.fronius.device","inverter","192.168.200.252","0","0","none","1"]
2022-05-17 07:10:31: PID: 2381: root: Fronius IP-Adresse: 192.168.200.252
2022-05-17 07:10:31: PID: 2381: root: Komponente Fronius Wechselrichter auslesen.
2022-05-17 07:10:31: PID: 2381: root: Fronius Wechselrichter: FaultState FaultStateLevel.ERROR, FaultStr <class 'KeyError'> 'ip_address', Traceback:
Traceback (most recent call last):
File "/var/www/html/openWB/packages/modules/fronius/device.py", line 111, in read_legacy
total_power = inverter1.read_power()
File "/var/www/html/openWB/packages/modules/fronius/inverter.py", line 40, in read_power
'http://' + self.device_config["ip_address"] + '/solar_api/v1/GetPowerFlowRealtimeData.fcgi',
KeyError: 'ip_address'

**Pull request merged**